client/shared/src/main/scala/org/http4s/client/websocket/WSClient.scala

```diff
@@ -179,21 +214,18 @@ private[http4s] object WSClient {
       override def connectHighLevel(request: WSRequest) =
         for {
           recvCloseFrame <- Resource.eval(Deferred[F, WSFrame.Close])
-          outputOpen <- Resource.eval(Ref[F].of(false))
```

Comment:
Was this a bug? Seems that it was never set to `true`.

Reply:
Oh, yeah, that is/was a bug. The idea was track whether we did send a close frame yet was to prevent the following situation, as it would result in an exception: We send a close frame via `sendClose`, then we receive a close frame, and then send another one in response to that. The current faulty implementation indeed does prevent this exception, but ofc simply never replies with a close frame...